userspace/libsinsp/plugin.cpp

```diff
@@ -712,6 +713,18 @@ void sinsp_plugin::validate_init_config_json_schema(std::string& config, std::st
 	}
 }
 
+void sinsp_plugin::set_config(const std::string& config)
+{
+	std::string conf = config;
+	validate_config(conf);
+
+	if(m_state && m_handle->api.set_config)
```

Comment:
```suggestion
	if(m_handle->api.set_config)
```
I don't think that having a non-NULL state should be a requirement. The API currently allows NULL states in theory, so let's make the plugin decide whether to accept it or not.